src/main/java/com/o19s/es/ltr/query/RankerQuery.java

```diff
@@ -245,12 +249,18 @@ public Explanation explain(LeafReaderContext context, int doc) throws IOExceptio
         }
 
         @Override
+        @SuppressWarnings("unchecked")
         public RankerScorer scorer(LeafReaderContext context) throws IOException {
             List<Scorer> scorers = new ArrayList<>(weights.size());
             List<DocIdSetIterator> subIterators = new ArrayList<>(weights.size());
             MutableSupplier<LtrRanker.FeatureVector> vectorSupplier = new MutableSupplier<>();
-            for (Weight weight : weights) {
+            for (int idx = 0; idx < weights.size(); idx++) {
                 Scorer scorer;
+                Query query = queries.get(idx);
+                Weight weight = weights.get(idx);
+                if (query instanceof Suppliers.MutableSupplierInterface) {
+                    ((Suppliers.MutableSupplierInterface<Supplier<LtrRanker.FeatureVector>>) query).set(vectorSupplier);
```

Comment:
not a big fan of mutating the query here...

What I've been experimenting with is a new "ltrRewrite" phase, see https://github.com/nomoa/elasticsearch-learning-to-rank/commit/6cd83de564c95c03ad8daf509d2527995c8cf7ee